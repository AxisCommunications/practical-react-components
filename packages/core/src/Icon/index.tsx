import React from 'react'

import styled from 'styled-components'
import { iconSize } from '../designparams'

type BaseElement = HTMLSpanElement
type BaseProps = React.HTMLAttributes<BaseElement>

/**
 * TODO: Add RTL-support for icon ? (https://material.io/design/usability/bidirectionality.html#mirroring-elements)
 */

const getSize = (size: string): string => {
  switch (size) {
    case 'small':
      return iconSize.small
    case 'medium':
      return iconSize.medium
    case 'large':
      return iconSize.large
    case 'extraLarge':
      return iconSize.extraLarge
    default:
      return iconSize.medium
  }
}

const IconContainer = styled.span<{ readonly size: string }>`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  pointer-events: none;
  overflow: hidden;
  vertical-align: middle;

  height: ${({ size }) => getSize(size)};
  width: ${({ size }) => getSize(size)};

  > svg {
    fill: currentColor;
    height: 100%;
    width: auto;
  }
`

export type IconType =
  | React.MemoExoticComponent<React.FC<React.SVGProps<SVGSVGElement>>>
  | React.FC<React.SVGProps<SVGSVGElement>>

export type IconSize = 'small' | 'medium' | 'large' | 'extraLarge'

interface IIconProps extends BaseProps {
  /**
   * `class` to be passed to the component.
   */
  readonly className?: BaseProps['className']
  /**
   * The icon element.
   */
  readonly icon: IconType
  /**
   * Changes the size of the Icons.
   */
  readonly size?: IconSize
}

export const Icon: React.FunctionComponent<IIconProps> = ({
  icon: SvgIcon,
  size = 'medium',
  ...props
}) => (
  <IconContainer role="img" size={size} {...props}>
    <SvgIcon />
  </IconContainer>
)

export const ClickableIcon = styled(Icon).attrs({ tabIndex: 0 })`
  cursor: pointer;
  pointer-events: auto;

  :focus {
    outline: 2px solid ${({ theme }) => theme.color.textPrimary()};
  }
`
