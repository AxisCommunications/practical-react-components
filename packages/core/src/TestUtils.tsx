import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { PracticalProvider } from './Practical'

/*
  Only used for Jest Snapshot Testing
*/

export const TP = ({
  children,
}: {
  readonly children: React.ReactElement
}): JSX.Element => <PracticalProvider>{children}</PracticalProvider>

/**
 * Mock reference to practical-root
 */
const createNodeMock = (element: React.ReactElement) => {
  if (element.props.id !== 'practical-root') {
    return null
  }

  return {}
}

export const TestRender = (jsx: JSX.Element) => {
  return renderer.create(<TP>{jsx}</TP>, { createNodeMock }).toJSON()
}

export const TestText = (): JSX.Element => <span>Test</span>

/* eslint-disable-next-line react/display-name */
export const TestTextWithForwardRef = React.forwardRef<HTMLElement>(
  ({ ...props }, ref) => (
    <span {...props} ref={ref}>
      Test
    </span>
  )
)
