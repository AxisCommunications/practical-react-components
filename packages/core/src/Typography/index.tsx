import styled, { css } from 'styled-components'

export type TypographyVariant =
  | 'chip-tag-text'
  | 'explanatory-text'
  | 'caption'
  | 'navigation-group'
  | 'navigation-label'
  | 'card-title'
  | 'group-title'
  | 'column-heading'
  | 'header-title'
  | 'default-text'
  | 'dialog-heading'
  | 'page-heading'
  | 'button-text'
  | 'compact-label'

const TYPOGRAPHY_VARIANTS = {
  /**
   * Paragraph
   *
   * The sort of text that appears in blocks inside
   * e.g. a modal, should be easily readable.
   */
  'default-text': {
    as: 'p',
    css: css`
      font-size: ${({ theme }) => theme.font.size.regular};
      line-height: ${({ theme }) => theme.font.lineHeight.large};
    `,
  },

  /**
   * Descriptive
   *
   * The sort of text that elaborates on another
   * piece of information or context.
   * Should be small to be able to show a lot.
   */
  caption: {
    as: 'p',
    css: css`
      font-size: ${({ theme }) => theme.font.size.small};
      line-height: ${({ theme }) => theme.font.lineHeight.small};
    `,
  },

  /**
   * Dialog heading
   */
  'dialog-heading': {
    as: 'p',
    css: css`
      font-size: ${({ theme }) => theme.font.size.larger};
      line-height: ${({ theme }) => theme.font.lineHeight.larger};
    `,
  },

  /**
   * Page heading
   */
  'page-heading': {
    as: 'p',
    css: css`
      font-size: ${({ theme }) => theme.font.size.larger};
      line-height: ${({ theme }) => theme.font.lineHeight.larger};
    `,
  },

  /**
   * Chip or tag text
   *
   * Text appearing as chip or tag.
   * Should be small text.
   */
  'chip-tag-text': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.small};
      line-height: ${({ theme }) => theme.font.lineHeight.smaller};
      white-space: nowrap;
    `,
  },

  /**
   * Explanatory text
   *
   * Text appearing as explanatory text, e.g. error message in a field.
   * Should be small text.
   */
  'explanatory-text': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.small};
      line-height: ${({ theme }) => theme.font.lineHeight.small};
      white-space: nowrap;
    `,
  },

  /**
   * Navigation group
   *
   * Text appearing as navigation group name, e.g. Group name on sidebar.
   */
  'navigation-group': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.small};
      line-height: ${({ theme }) => theme.font.lineHeight.smaller};
      white-space: nowrap;
    `,
  },

  /**
   * Navigation label
   *
   * Standard size used for text labels,
   * short descriptions connected to another element.
   * e.g. Label on sidebar
   */
  'navigation-label': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.regular};
      line-height: ${({ theme }) => theme.font.lineHeight.small};
      white-space: nowrap;
    `,
  },

  /**
   * Card title
   */
  'card-title': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.regular};
      font-weight: ${({ theme }) => theme.font.fontWeight.semibold};
      line-height: ${({ theme }) => theme.font.lineHeight.small};
      white-space: nowrap;
    `,
  },

  /**
   * Group title
   *
   * Text appearing as a title of some fields or some group in Dialog.
   */
  'group-title': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.regular};
      font-weight: ${({ theme }) => theme.font.fontWeight.semibold};
      line-height: ${({ theme }) => theme.font.lineHeight.small};
      white-space: nowrap;
    `,
  },

  /**
   * Column heading in a table
   */
  'column-heading': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.small};
      font-weight: ${({ theme }) => theme.font.fontWeight.normal};
      line-height: ${({ theme }) => theme.font.lineHeight.smaller};
      white-space: nowrap;
    `,
  },

  /**
   * Header title
   */
  'header-title': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.regular};
      font-weight: ${({ theme }) => theme.font.fontWeight.semibold};
      line-height: ${({ theme }) => theme.font.lineHeight.small};
      white-space: nowrap;
    `,
  },

  /**
   * Button
   *
   * Text appearing inside a button
   */
  'button-text': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.large};
      font-weight: ${({ theme }) => theme.font.fontWeight.semibold};
      line-height: ${({ theme }) => theme.font.lineHeight.large};
      white-space: nowrap;
    `,
  },

  /**
   * Compact label
   *
   * Label where space is limited
   */
  'compact-label': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.smaller};
      line-height: ${({ theme }) => theme.font.lineHeight.smallest};
    `,
  },
}

/**
 *
 * Generic typography
 *
 */

export interface TypographyProps {
  readonly variant?: TypographyVariant
}

export const Typography = styled.p.attrs<TypographyProps>(
  ({ variant = 'default-text' }) => ({
    as: TYPOGRAPHY_VARIANTS[variant].as,
  })
)<TypographyProps>`
  font-family: ${({ theme }) => theme.font.family};
  font-size: ${({ theme }) => theme.font.size.regular};
  line-height: ${({ theme }) => theme.font.lineHeight.large};
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 0;

  ${({ variant = 'default-text' }) => TYPOGRAPHY_VARIANTS[variant].css}
`
