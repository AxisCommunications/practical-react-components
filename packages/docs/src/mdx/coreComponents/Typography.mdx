export const meta = {
  name: 'Typography âœ”',
  route: '/styles/typography',
  menu: 'Styles',
}

import {
  SimpleTable,
  SimpleTableRow,
  Typography,
} from 'practical-react-components-core'

# Typography

## Design

Typography is used to present the design and content as clearly and efficiently as possible.
It contains reusable categories of text, each with an intended application and meaning.

<img
  src={require('../images/typographies.png').default}
  alt="Typography design"
/>

## Demo

export const TABLE_HEADER_DATA = [
  'Variant',
  'Font size',
  'Case with actual size',
]

export const TYPOGRAPHY_LIST = [
  { variant: 'page-heading', fontSize: '21px', case: 'Page heading' },
  { variant: 'dialog-heading', fontSize: '21px', case: 'Dialog heading' },
  {
    variant: 'button-text',
    fontSize: '14px',
    case: 'Regular button text',
  },
  {
    variant: 'default-text',
    fontSize: '13px',
    case: `Text in a paragraph. Lorem ipsum dolor sit amet, consectetur adipiscing elit,
  sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.`,
  },
  {
    variant: 'header-title',
    fontSize: '13px',
    case: 'Header title',
  },
  {
    variant: 'group-title',
    fontSize: '13px',
    case: 'Group title',
  },
  {
    variant: 'card-title',
    fontSize: '13px',
    case: 'Card title',
  },
  {
    variant: 'navigation-label',
    fontSize: '13px',
    case: 'A navigation label belong to something',
  },
  {
    variant: 'column-heading',
    fontSize: '12px',
    case: 'Use me for table headers',
  },
  {
    variant: 'navigation-group',
    fontSize: '12px',
    case: 'Navigation group',
  },
  {
    variant: 'caption',
    fontSize: '12px',
    case: 'Extra descriptive text used as a caption.',
  },
  {
    variant: 'explanatory-text',
    fontSize: '12px',
    case: 'A explanatory text, e.g. error message in a field. ',
  },
  {
    variant: 'chip-tag-text',
    fontSize: '12px',
    case: 'Piece of data in e.g. a chip or a tag',
  },
  {
    variant: 'compact-label',
    fontSize: '11px',
    case: 'Some label where space is limited',
  },
]

<SimpleTable
  maxHeight="400"
  header={TABLE_HEADER_DATA.map((cell, index) => (
    <Typography variant="column-heading" key={index}>
      {cell}
    </Typography>
  ))}
>
  {TYPOGRAPHY_LIST.map((typography, index) => (
    <SimpleTableRow key={index}>
      <Typography>{typography.variant}</Typography>
      <Typography>{typography.fontSize}</Typography>
      <Typography variant={typography.variant}>{typography.case}</Typography>
    </SimpleTableRow>
  ))}
</SimpleTable>

## Basic usage

```typescript type=live
<Typography variant="column-heading">Text</Typography>
```

## Props

Typography forwards all unknown props to the styled-components component. See
[`API Reference`](https://styled-components.com/docs/api) for more
details on which additional props that are available.

<Props of="Typography" />
