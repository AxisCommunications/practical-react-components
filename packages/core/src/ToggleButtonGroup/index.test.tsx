import React, { useState, useCallback } from 'react'
import { LocationIcon, CalendarIcon } from 'practical-react-components-icons'
import 'jest-styled-components'

import { ToggleButtonGroup, ToggleButtonGroupWithField } from '.'
import { TestRender } from '../TestUtils'
import { Typography } from '../Typography'
import { Icon } from '../Icon'

const TestController: React.FC = () => {
  const [values, setToggledValues] = useState(['24'])
  const onChange = useCallback(
    id => {
      const found = values.includes(id)
      if (found) {
        setToggledValues(values.filter(v => v !== id))
      } else {
        setToggledValues([...values, id])
      }
    },
    [values]
  )
  return (
    <ToggleButtonGroup
      onChange={onChange}
      options={[
        {
          id: '24',
          content: <Typography variant="default-text">24</Typography>,
        },
        { id: 'location', content: <Icon icon={LocationIcon} /> },
        {
          id: 'disabled',
          content: <Typography variant="default-text">Disabled</Typography>,
          disabled: true,
        },
      ]}
      values={values}
    />
  )
}

const TestControllerWithField: React.FC = () => {
  const [values, setToggledValues] = useState(['24'])
  const onChange = useCallback(
    id => {
      const found = values.includes(id)
      if (found) {
        setToggledValues(values.filter(v => v !== id))
      } else {
        setToggledValues([...values, id])
      }
    },
    [values]
  )
  return (
    <ToggleButtonGroupWithField
      label="Togglebutton test"
      onChange={onChange}
      options={[
        {
          id: 'btn',
          content: <Typography variant="default-text">click me</Typography>,
        },
        {
          id: 'calendar',
          content: <Icon icon={CalendarIcon} />,
        },
        {
          id: 'test',
          content: <Typography variant="default-text">test</Typography>,
        },
      ]}
      values={values}
    />
  )
}

describe('ToggleButtonGroup', () => {
  test('ToggleButtonGroup', () => {
    const test = TestRender(<TestController />)
    expect(test).toMatchSnapshot()
  })
  test('ToggleButtonGroupWithField', () => {
    const test = TestRender(<TestControllerWithField />)
    expect(test).toMatchSnapshot()
  })
})
