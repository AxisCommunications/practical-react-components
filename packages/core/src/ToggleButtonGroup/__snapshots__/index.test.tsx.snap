// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButtonGroup ToggleButtonGroup 1`] = `
.c5 {
  font-family: Open Sans,Helvetica,arial,sans-serif;
  font-size: 13px;
  line-height: 22px;
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 0;
  font-size: 13px;
  line-height: 22px;
}

.c3 {
  box-sizing: border-box;
  max-width: 100%;
  position: relative;
  cursor: pointer;
  -webkit-transition: all 200ms;
  transition: all 200ms;
}

.c3:not(:last-child) {
  border-right: 1px solid rgb(184,184,184);
}

.c3:active {
  z-index: 1;
  box-shadow: -1px 0 0 5px rgba(103,58,183,0.24);
  border-radius: 4px;
}

.c6 {
  box-sizing: border-box;
  max-width: 100%;
  position: relative;
  cursor: pointer;
  -webkit-transition: all 200ms;
  transition: all 200ms;
}

.c6:not(:last-child) {
  border-right: 1px solid rgb(184,184,184);
}

.c9 {
  box-sizing: border-box;
  max-width: 100%;
  position: relative;
  cursor: pointer;
  -webkit-transition: all 200ms;
  transition: all 200ms;
  opacity: 0.48;
  pointer-events: none;
  cursor: default;
  box-shadow: none;
}

.c9:not(:last-child) {
  border-right: 1px solid rgb(184,184,184);
}

.c4 {
  background-color: rgb(255,255,255);
  height: 32px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 8px;
  overflow: hidden;
  white-space: nowrap;
  -webkit-transition: all 200ms;
  transition: all 200ms;
  color: rgb(255,255,255);
  fill: rgb(255,255,255);
  background-color: rgb(103,58,183);
}

.c4:focus {
  outline-color: rgb(69,39,160);
}

.c4:hover {
  background-color: rgb(69,39,160);
}

.c7 {
  background-color: rgb(255,255,255);
  height: 32px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 8px;
  overflow: hidden;
  white-space: nowrap;
  -webkit-transition: all 200ms;
  transition: all 200ms;
  color: rgb(102,102,102);
  fill: rgb(102,102,102);
}

.c7:hover {
  background-color: rgba(204,204,204,0.16);
}

.c7:focus {
  background-color: rgba(204,204,204,0.16);
  outline-color: rgb(69,39,160);
}

.c7:active {
  background-color: rgba(204,204,204,0.24);
}

.c2 {
  width: 100%;
  display: grid;
  grid-template-columns: repeat(auto-fit,minmax(24px,1fr));
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid rgb(184,184,184);
  border-radius: 2px;
}

.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  overflow: hidden;
  vertical-align: middle;
  height: 24px;
  width: 24px;
}

.c8 > svg {
  fill: currentColor;
  height: 100%;
  width: auto;
}

.c11 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c10 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-1"
    >
      <div
        className="c2"
      >
        <div
          className="c3"
          disabled={false}
          onClick={[Function]}
          selected={true}
        >
          <div
            className="c4"
            selected={true}
            tabIndex={1}
          >
            <p
              className="c5"
            >
              24
            </p>
          </div>
        </div>
        <div
          className="c6"
          disabled={false}
          onClick={[Function]}
          selected={false}
        >
          <div
            className="c7"
            selected={false}
            tabIndex={1}
          >
            <span
              className="c8"
              role="img"
              size="medium"
            >
              <svg
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          className="c9"
          disabled={true}
          onClick={[Function]}
          selected={false}
        >
          <div
            className="c7"
            selected={false}
            tabIndex={1}
          >
            <p
              className="c5"
            >
              Disabled
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="c10"
      id="layer-2"
    >
      <div
        className="c11"
      />
    </div>
  </div>
</div>
`;

exports[`ToggleButtonGroup ToggleButtonGroupWithField 1`] = `
.c3 {
  font-family: Open Sans,Helvetica,arial,sans-serif;
  font-size: 13px;
  line-height: 22px;
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 0;
  font-size: 13px;
  line-height: 18px;
  white-space: nowrap;
}

.c7 {
  font-family: Open Sans,Helvetica,arial,sans-serif;
  font-size: 13px;
  line-height: 22px;
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 0;
  font-size: 13px;
  line-height: 22px;
}

.c2 {
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgb(82,82,82);
  cursor: default;
}

.c5 {
  box-sizing: border-box;
  max-width: 100%;
  position: relative;
  cursor: pointer;
  -webkit-transition: all 200ms;
  transition: all 200ms;
}

.c5:not(:last-child) {
  border-right: 1px solid rgb(184,184,184);
}

.c6 {
  background-color: rgb(255,255,255);
  height: 32px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 8px;
  overflow: hidden;
  white-space: nowrap;
  -webkit-transition: all 200ms;
  transition: all 200ms;
  color: rgb(102,102,102);
  fill: rgb(102,102,102);
}

.c6:hover {
  background-color: rgba(204,204,204,0.16);
}

.c6:focus {
  background-color: rgba(204,204,204,0.16);
  outline-color: rgb(69,39,160);
}

.c6:active {
  background-color: rgba(204,204,204,0.24);
}

.c4 {
  width: 100%;
  display: grid;
  grid-template-columns: repeat(auto-fit,minmax(24px,1fr));
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid rgb(184,184,184);
  border-radius: 2px;
}

.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  overflow: hidden;
  vertical-align: middle;
  height: 24px;
  width: 24px;
}

.c8 > svg {
  fill: currentColor;
  height: 100%;
  width: auto;
}

.c10 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c9 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-3"
    >
      <div>
        <div
          className="c2"
        >
          <span
            className="c3"
          >
            Togglebutton test
          </span>
        </div>
        <div
          className="c4"
        >
          <div
            className="c5"
            disabled={false}
            onClick={[Function]}
            selected={false}
          >
            <div
              className="c6"
              selected={false}
              tabIndex={1}
            >
              <p
                className="c7"
              >
                click me
              </p>
            </div>
          </div>
          <div
            className="c5"
            disabled={false}
            onClick={[Function]}
            selected={false}
          >
            <div
              className="c6"
              selected={false}
              tabIndex={1}
            >
              <span
                className="c8"
                role="img"
                size="medium"
              >
                <svg
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            className="c5"
            disabled={false}
            onClick={[Function]}
            selected={false}
          >
            <div
              className="c6"
              selected={false}
              tabIndex={1}
            >
              <p
                className="c7"
              >
                test
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="c9"
      id="layer-4"
    >
      <div
        className="c10"
      />
    </div>
  </div>
</div>
`;
